sgt-back-end/index.js

```diff
@@ -0,0 +1,187 @@
+      });
+    });
+})
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+})
+
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name, course = req.body.course, score = Number(req.body.score);
+  if (!name || !course || (!score && score != 0)) {
+    return res.status(400).json({
+      error: '"name", "course", and "score" are required fields.'
+    })
+  }
+  if (!Number.isInteger(score) || score < 0 || score > 100) {
+    return res.status(400).json({
+      error: '"score" must be an integer value between 0-100.'
+    })
+  }
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values ($1, $2, $3)
+    returning *;
+  `;
+  const params = [name, course, score];
+  db.query(sql, params)
+    .then(result => {
+      const newGrade = result.rows[0];
+      res.status(201).json(newGrade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured.'
+      })
+    })
+})
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const gradeId = Number(req.params.gradeId);
+  const name = req.body.name, course = req.body.course, score = Number(req.body.score);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+  if (!name || (!score && score != 0) || !course) {
+    return res.status(400).json({
+      error: '"name", "course", and "score" are required fields.'
+    })
+  }
+  if (!Number.isInteger(score) || score < 0 || score > 100) {
+    return res.status(400).json({
+      error: '"score" must be an integer value between 0-100.'
+    })
+  }
+  const sql = `
+    update "grades"
+    set "name" = $2,
+        "course" = $3,
+        "score" = $4
+    where "gradeId" = $1
+    returning *;
+  `;
+  const params = [gradeId, name, course, score];
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    })
+})
+
+app.delete('/api/grades/:gradeId', (req,res) => {
+  const gradeId = parseInt(req.params.gradeId, 10);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
```

Comment:
This should reject all `gradeId`s which aren't whole numbers, not silently fix them